src/azure-cli/azure/cli/command_modules/appconfig/_kv_helpers.py

```diff
@@ -1044,15 +1057,20 @@ def __import_kvset_from_file(client, path, yes):
         if __validate_import_config_setting(config_setting):
             kvset_to_import.append(config_setting)
 
+    existing_kvset = __read_kv_from_config_store(client,
+                                                 key=SearchFilterOptions.ANY_KEY,
+                                                 label=SearchFilterOptions.ANY_LABEL)
```

Comment:
This should be under ``` if strict```

Reply:
We need `existing_kvset` regardless of whether it is strict or not.
The only difference is when it is not strict, we the filter the set to not include missing kvs. (line #1073)

Reply:
No. We only need to fetch existing kvsets if strict or preview. Otherwise, we are loading the whole store for every single kvset import.